### ISSUE TITLE
DA notebook

### ISSUE THREAD

**Issue by jgomezdans** [UCL-EO/Workshop2022#15]
- [ ] Model exploration with parameters ("sliders")
- [ ] Automatically fit the LAI ("DA button")
- [ ] Per pixel LAI exploration
    - [ ] Show yield
    - [ ] Show parameter spatial distribution

**Comment by jgomezdans** [comment]
Need to update GUI and test "DA button" with big safe ensemble Tuesday.

**Comment by jgomezdans** [comment]
I have tested the DA button, but my results are very different to yours @Ma-hy
![image](https://user-images.githubusercontent.com/139304/156233269-f330f347-8882-46c8-bca1-18f17b5eb442.png)

Looking into your ensemble fitting code, @MarcYin  noticed that you calculate the wrong LAI value:
```
np.nanmean(np.transpose(obs['mean_bios_all'][:,4,:])*obs['mean_bio_scales_all'][:,4], axis=1)
```
the LAI should be calculated as 
```
f.f.mean_bios_all[:, 4, :].mean(axis=0)
```

We need to discuss this tomorrow morning 9am. I've sent an invite through.

**Comment by profLewis** [comment]
Also, can you do plots of the parameters? We surely need to see these to be able to interpret

Lewis

Professor Lewis, NCEO/UCL
________________________________

**Comment by Ma-hy** [comment]
Tried assimilating correct LAI, like below
![image](https://user-images.githubusercontent.com/17977287/156369280-8e293b5d-8dda-44d3-8573-bb3c1f850c6a.png)

**Comment by Ma-hy** [comment]
Compared with the wrong LAI calculation
![image](https://user-images.githubusercontent.com/17977287/156369735-0e9d7bd5-7c8b-4952-8be4-6981de3e0a84.png)

**Comment by jgomezdans** [comment]
Your new thing is the same as my plot above: only a small variation between 2200-2500 kg/ha. Plot them with the same axes.

**Comment by Ma-hy** [comment]
Yeah, the slope is quite flat, but keeps high R value, by the way, the pixel-by-pixel assimilation shows a bit larger variation.
![image](https://user-images.githubusercontent.com/17977287/156372121-d326f466-b552-4427-b63b-41b569325a9e.png)

**Comment by Ma-hy** [comment]
![image](https://user-images.githubusercontent.com/17977287/156375772-f2b64fbd-41a2-498e-a44e-8b3521efd702.png)
Yeah, the slope is quite flat, but keeps high R value, by the way, the pixel-by-pixel assimilation shows a bit larger variation.

**Comment by profLewis** [comment]
So what is the way forward here? It sound like you don’t have a viable DA for the maize crop. Do we have a viable workshop then?

Did you try filtering from the empirical relationship?

Lewis

Professor Lewis, NCEO/UCL
________________________________

**Comment by jgomezdans** [comment]
I'm currently filtering by the empirical relationship and will have some results in a bit. the ensembles from Hongyuan didn't cover empirical relationship line, so have changed things a bit to get to that point.

**Comment by jgomezdans** [comment]
OK, so after fitting the empirical line, and changing a more sensible set of parameters, here's something more hopeful. Results first:
* Fit to the LAI is fair. Remember, small ensemble (made smaller by filtering, around 650 ensemble members)
![image](https://user-images.githubusercontent.com/139304/156418421-b1b473ef-5dc5-4412-bb0b-6ec3dafb2a7a.png)
* Observed vs predicted yield: Pretty good correlation, slope around 0.45
![image](https://user-images.githubusercontent.com/139304/156418632-a9c819bc-d8b2-4951-b04a-619ce82ccae9.png)
* Parameters vs obs yield. Reasonable trends (see for example CVO), but there's a lot of internal correlation here
![image](https://user-images.githubusercontent.com/139304/156419001-41116b82-7947-4c0b-8519-6783ef45bd12.png)
* Correlation matrix shows huge correlations, so we may be able to further simplify the number of crop parameters 
![image](https://user-images.githubusercontent.com/139304/156419976-2fa53029-667b-47b7-ba25-8534fc448ea5.png)

I'm doing this on a small ensemble, so I selected the 5 best solutions (for the correlation matrix, I selected the best 20). In general, this looks reasonable, and I think we can maybe expect better results with bigger ensembles that allow better fitting to LAI, and possibly reducing the input parameters space (not too bothered about that).

**Comment by MarcYin** [comment]
Assimilation code is working OK for fitting LAI etc, but using the DA parameters to forward model observations doesn't produce the same as the ensemble. Likely to be a discrepancy between ensemble generation code and notebook code.

![Screenshot 2022-03-03 at 19 35 08](https://user-images.githubusercontent.com/19539641/156639644-84fd260c-7c56-439c-98f0-746c32c7d8e0.png)
s

**Comment by jgomezdans** [comment]
Mostly there, bits missing are showing the per pixel DA results.
Additionally, @Ma-hy has found a new smaller set of parameters for the model, that are maybe worthwhile exploring, but we'll see in discussion.

**Comment by profLewis** [comment]
Can you show the growth stage on the graphs? To give some context to the amax parameters

Professor Lewis, NCEO/UCL
________________________________

**Comment by jgomezdans** [comment]
It's one of the drop-down parameters, called DVS. Probably should use the full names and not the abbreviations
________________________________

**Comment by profLewis** [comment]
Looking good people. Thanks.

I’d very much like to know more about what the empirical fn filtering did you the prior parameter space. As I said, I think this is pragmatic and seems to get us a solution for the workshop. But it might have other important implications for how to approach this.

Without that constraint, it seems to be suggesting that fitting lai alone can’t get us to yield using DA. I know there are some issues with much of the crop DA work out there, but that’s quite an important issue. Or maybe this is just a special case, since the drivers are so coarse and effectively the same for all fields, but the yield vary so much.

Does the empirical constraint filtering just put the prior in a more sensible place? Or … are we just really doing no more than mimicking the empirical relationship here?

Lewis

Professor Lewis, NCEO/UCL
________________________________
